hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -2903,6 +2921,47 @@ func computeUnmanagedEtcdAvailability(hcluster *hyperv1.HostedCluster, unmanaged
 	}
 }
 
+func computeAWSEndpointServiceCondition(awsEndpointServiceList hyperv1.AWSEndpointServiceList, conditionType hyperv1.ConditionType) metav1.Condition {
```

Comment:
this new logic deserves a unit test
- when both condition A and B are true
- when A is true and B is false 
- when A is false and B is true
- when both condition A and B are false